src/simulator/simulator.py

```diff
@@ -6,44 +6,76 @@
 WORLD_HEIGHT = 160
 
 import pygame
+from copy import deepcopy
 import numpy as np
 from classes.vehicle import Vehicle
+from classes.button import Button
 from manager.manager import Manager
 from manager.route import Node, Edge, Route
 from .render import render_world, render_manager
 from .update import update_world
 
+def vehicle_copy(vehicles: list[Vehicle]) -> list[Vehicle]:
+    return [Vehicle(v.id, v.position, v.velocity, v.acceleration, v.direction, v.width, v.length, v.pivot_distance, v.intent, v.image_source) for v in vehicles]
+
```

Comment:
- [ ] I think this may be a helper function belonging in helper, or even somewhere else

Reply:
I agree, perhaps in the Vehicle Class?